In [1]:
import seaborn as sns
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
from scripts.result_parser import get_df
%matplotlib inline

## Free energy of Zika virus

Next, we present a more challenging problem that studies the free energy of the Zika virus.
We acquired the molecular structure (6CO8) from ?, parameterized it with pdb2pqr and generated mesh on the solvent-excluded surface using Nanoshaper.
The prepared structure contains about 1.6 million atoms (charges) and our mesh has around 10 million boundary elements, which corresponds to an element density of ?.
In this experiment, 3 quadrature points were used for regular Galerkin integrals over disjoint elements. The FMM expansion order was 4 and the tolerance of GMRES was $10^{-4}$.

Table ? shows the result and performance using both formulations.

In [2]:
direct_6CO8_df = get_df('runs/6CO8/direct/', formulation='direct', skip4=True)
juffer_6CO8_df = get_df('runs/6CO8/juffer/', formulation='juffer', skip4=True)
zika_df = pd.concat([direct_6CO8_df, juffer_6CO8_df])
zika_df.index = ['direct', 'Juffer']
zika_df

t_total_assembly  t_total_gmres  num_iter  e_solv [kcal/Mol]  \
direct       1505.478204    9359.422097       105     -116587.549968   
Juffer       7180.632862    9510.991671        37     -116190.917437   

        memory [GB]  t_fmm_init  t_singular_assembler  t_assemble_sparse  \
direct   111.834920  143.466971            465.879339              14.27   
Juffer   172.844268  162.679122            665.395250              96.15   

        t_assembly_other  t_singular_correction    t_laplace  t_helmholtz  \
direct        881.861894             610.815506  2109.317258  5738.135942   
Juffer       6256.408490             532.724651  1543.242236  5731.557996   

        t_avg_laplace  t_avg_helmholtz  t_gmres_other  
direct       4.928311        13.406860     901.153392  
Juffer       4.946289        13.360275    1703.466788

-------

### Appendix:

1. results using various regular quadrature order

| quadrature order | expansion order | tol  | e_solv     | gmres time (s) | # iter |
|:----------------:|-----------------|------|------------|----------------|--------|
|         1        | 4               | 1e-4 | -123693.42 | 4149.30        | 105    |
|         2        | 4               | 1e-4 | -116587.55 | 10786.23       | 105    |
|         3        | 4               | 1e-4 | -116491.99 | 13273.08       | 105    |
|         4        | 4               | 1e-4 | -116746.19 | 20891.13       | 105    |
|         4        | 5               | 1e-4 | -116676.47 | 75556.11       | 320    |

2. result from PyGBe: -117261.125498 kcal/mol